qubes/ext/core_features.py

```diff
@@ -62,3 +62,20 @@ def qubes_features_request(self, vm, event, untrusted_features):
             # if this is the first time qrexec was advertised, now can finish
             #  template setup
             yield from vm.fire_event_async('template-postinstall')
+
+    # pylint: disable=no-self-use
+    def set_servicevm_feature(self, subject):
+        if subject.provides_network:
+            subject.features['servicevm'] = 1
+        elif 'servicevm' in subject.features:
+            del subject.features['servicevm']
+
+    @qubes.ext.handler('property-set:provides_network')
+    def on_property_set(self, subject, event, name, newvalue, oldvalue=None):
+        # pylint: disable=unused-argument
+        self.set_servicevm_feature(subject)
+
+    @qubes.ext.handler('property-del:provides_network')
+    def on_property_del(self, subject, event, name):
+        # pylint: disable=unused-argument
+        self.set_servicevm_feature(subject)
```

Comment:
This all will work fine for new VMs, because default value for provides_network is false (so default not having `servicevm` feature is correct).
But the above miss adding the feature for existing VMs (on system update from the version before this change). The easiest way to do that is adding similar handler for `domain-load` event (loading a domain from qubes.xml - basically qubesd service start).